src/radical/pilot/pmgr/launching/default.py

```diff
@@ -888,42 +890,51 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand, tar_name):
             if virtenv_mode != 'private':
                 cleanup = cleanup.replace('v', '')
 
-        # use local VE ?
-        if virtenv_mode == 'local':
-            if os.environ.get('VIRTUAL_ENV'):
-                python_dist = 'default'
-                virtenv     = os.environ['VIRTUAL_ENV']
-            elif os.environ.get('CONDA_PREFIX'):
-                python_dist = 'anaconda'
-                virtenv     = os.environ['CONDA_PREFIX']
-            else:
-                # we can't use local
-                self._log.error('virtenv_mode is local, no local env found')
-                raise ValueError('no local env found')
+        # estimate requested resources
+
+        if os.environ.get('RADICAL_SAGA_SMT'):
+            try:
+                system_architecture['smt'] = int(os.environ['RADICAL_SAGA_SMT'])
+            except Exception as e:
+                self._log.debug('SAGA SMT not set: %s' % e)
```

Comment:
maybe use `self._log.exception`?

Reply:
I think we can skip `try..except` and follow your proposal above
(btw `debug` was used since it wasn't a required parameter, and we didn't enforce it)